clients/cloud/rust/src/consumer.rs

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+extern crate futures;
+extern crate rdkafka;
+extern crate tokio;
+
+use futures::stream::Stream;
+use rdkafka::consumer::stream_consumer::StreamConsumer;
+use rdkafka::consumer::Consumer;
+use rdkafka::Message;
+use std::boxed::Box;
+use tokio::runtime::current_thread::Runtime;
+
+mod utils;
+
+fn echo_message<M: Message>(msg: M) -> Result<(), std::str::Utf8Error> {
+    let deserialize = |o| match o {
+        None => Ok("".to_string()),
+        Some(val) => Ok(std::str::from_utf8(val)?.to_string()),
+    };
+
+    println!(
+        "Consumed record from topic {} partition [{}] @ offset {} with key {} and value {}",
+        msg.topic(),
+        msg.partition(),
+        msg.offset(),
+        deserialize(msg.key())?,
+        deserialize(msg.payload())?,
+    );
+
+    Ok(())
+}
+
+fn main() -> Result<(), Box<std::error::Error>> {
```

Comment:
Should this be `dyn std::error::Error`?

Reply:
I got the inspiration from [here](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/question-mark-in-main-and-tests.html).